cs/src/core/Bond.cs

```diff
@@ -3,7 +3,52 @@
 
 namespace Bond
 {
-    using System.Collections.Generic;
+    using System;
+    using System.Collections.Generic;
+
+    public partial class GUID
+    {
+        public static implicit operator Guid(GUID bondGuid)
+        {
+            return BitConverter.IsLittleEndian ?
```

Comment:
Can't use Assert.IsTrue in this context.
Can only use:
Debug.Assert(BitConverter.IsLittleEndian, "GUID conversion is only little endian for now");
Is this fine to have? Otherwise if bond doesn't support BigEndian than more should break than just this so we may be able to do without the check altogether